src/pushNotification.ts

```diff
@@ -0,0 +1,111 @@
+import HathorWalletServiceWallet from './wallet/wallet';
+import { axiosInstance } from './wallet/api/walletServiceAxios';
+import { WalletRequestError } from './errors';
+
+export class PushNotification {
+
+  /**
+   * Register the device to receive push notifications
+   *
+   * @memberof PushNotification
+   * @inner
+   */
+  static async registerDevice(wallet: HathorWalletServiceWallet, payload: PushRegisterRequestData): Promise<PushNotificationResult> {
+    wallet.failIfWalletNotReady();
+    const data = await walletServiceClient.pushRegister(wallet, payload);
+    return data;
+  }
+
+  /**
+   * Update the device settings to receive push notifications
+   *
+   * @memberof PushNotification
+   * @inner
+   */
+  static async updateDevice(wallet: HathorWalletServiceWallet, payload: PushUpdateRequestData): Promise<PushNotificationResult> {
+    wallet.failIfWalletNotReady();
+    const data = await walletServiceClient.pushUpdate(wallet, payload);
+    return data;
+  }
+
+  /**
+   * Delete the device from the push notification service
+   *
+   * @memberof PushNotification
+   * @inner
+   */
+  static async unregisterDevice(wallet: HathorWalletServiceWallet, deviceId: string): Promise<PushNotificationResult> {
+    wallet.failIfWalletNotReady();
+    const data = await walletServiceClient.pushUnregister(wallet, deviceId);
+    return data;
+  }
+}
+
+const walletServiceClient = {
```

Comment:
I see 2 clients there: the public one `PushNotification` and the private one `walletServiceClient`. Despite they are part of the same process, they can vary in their behavior. This segregation let us evolve them at different paces because they have different needs. In the end, it also facilitates tests, and turn maintenance easier (imo).